blogs/2022/08/16/markdown-language-server.md

```diff
@@ -0,0 +1,100 @@
+---
+Order: 75
+TOCTitle: Markdown Language Server
+PageTitle: Introducing the Markdown Language Server
+MetaDescription: Why we decided to build a language server for working with Markdown
+Date: 2022-08-16
+Author: Matt Bierner
+---
+
+# Introducing the Markdown Language Server
+
+August 16, 2022 by Matt Bierner, [@MattBierner](https://twitter.com/MattBierner)
+
+Markdown was the first feature I took ownership of when I joined VS Code back in 2016. Wow, has it really been six years? It was a great match though. I've worked with Markdown long enough that I ever find myself hopefully typing backticks and asterisks into Twitter, Outlook, and just about every textbox my cursor lands in. It's been incredibly rewarding to grow VS Code's built-in Markdown support over the years and see how our Markdown extension has directly and indirectly shaped core features like webviews and notebooks.
+
+That's why I'm excited to share a project I've been quietly working towards for the past half year, and a project that I think represents the next step for VS Code's Markdown tooling: a Markdown Language Server. With this language server, we're making much of VS Code's built-in Markdown language tooling — everything from document outlines, to smart folding, to path completions — available to other editors and tools. Our goal is to push Markdown tooling forward with the type of smarts more often associated with programming languages.
+
+The Markdown Language Server effort is split between two new (and somewhat confusingly similarly named!) open source libraries:
+
+- [Markdown Language Service][service] — A TypeScript library that provides tools for working with Markdown.
+
+- [Markdown Language Server][server] — A [language server][ls] for Markdown built using the language service.
+
+While these libraries are still in early stages, they are already being used by VS Code 1.70+ (and hopefully you never even noticed :). We've even seen a few benefits from this switch, such as moving Markdown tooling to a separate process so it can't block other extensions. I'm looking forward to onboarding other editors and tools to these projects as well!
```

Comment:
missing nose dash and closing parenthesis? ':-))'

Reply:
I'm also wondering if we should leave out "I'm looking forward to onboarding other editors and tools....as well!" since it is talked about later in the post (more of a reveal :-) ).